dask/array/reductions.py

```diff
@@ -527,12 +533,27 @@ def nanmin(a, axis=None, keepdims=False, split_every=None, out=None):
     )
 
 
+def _nanmin_skip(x_chunk, axis, keepdims):
+    if len(x_chunk):
+        return np.nanmin(x_chunk, axis=axis, keepdims=keepdims)
+    else:
+        return asarray_safe(
+            np.array([], dtype=x_chunk.dtype), like=meta_from_array(x_chunk)
+        )
+
+
 @derived_from(np)
 def nanmax(a, axis=None, keepdims=False, split_every=None, out=None):
+    if np.isnan(a.size):
+        a = a.compute_chunk_sizes()
```

Comment:
Rather than computing chunk sizes implicitly, we should raise (or at least warn) for this case. You can import and use this message:

https://github.com/dask/dask/blob/fd8853952d030f3f63721520361fafdba65a12fd/dask/array/core.py#L83-L91

Reply:
Hi @jsignell 
Thanks for the review!
I totally agree that doing this quietly is not good, I am not sure if we should warn or raise an error.
Warning would change the behavior under the hood for users and they may not notice the warning.
Raising could brake user code but I guess that at least there is an easy fix with instructions in the error.
For now I changed it to raise.